Slack.Automation/Promact.Core.Repository/ScrumRepository/ScrumBotRepository.cs

```diff
@@ -390,374 +590,608 @@ public class ScrumBotRepository : IScrumBotRepository
             if (scrumStatus == ScrumStatus.NotStarted)
             {
-                Question question = questionList.FirstOrDefault();
-                Scrum scrum = new Scrum();
-                scrum.CreatedOn = DateTime.UtcNow;
-                scrum.GroupName = groupName;
-                scrum.ScrumDate = DateTime.UtcNow.Date;
-                scrum.ProjectId = project.Id;
-                scrum.TeamLeaderId = project.TeamLeaderId;
-                scrum.IsHalted = false;
-                scrum.IsOngoing = true;
-                _scrumRepository.Insert(scrum);
-
-                User firstEmployee = employees.FirstOrDefault();
-                //first employee is asked questions along with the previous day status (if any)
-                SlackUserDetails slackUser = await _slackUserDetails.GetByIdAsync(firstEmployee.SlackUserId);
-                replyMessage = _stringConstant.GoodDay + "<@" + slackUser.Name + ">!\n" + FetchPreviousDayStatus(firstEmployee.Id, project.Id) + question.QuestionStatement;
+                Question question = questionList.First();
+                User firstUser = users.FirstOrDefault(x => x.IsActive);
+                if (firstUser != null)
+                {
+                    SlackUserDetailAc slackUserDetailAc = await _slackUserDetailRepository.GetByIdAsync(firstUser.SlackUserId);
+                    if (slackUserDetailAc == null)
+                    {
+                        SlackUserDetails slackUserDetail = _slackUserDetails.FirstOrDefault(x => users.Where(y => y.IsActive).Select(y => y.SlackUserId).Contains(x.UserId));
+                        if (slackUserDetail != null)
+                            firstUser = users.FirstOrDefault(x => x.SlackUserId == slackUserDetail.UserId);
+                        else
+                            return _stringConstant.NoEmployeeFound;
+                    }
+                    Scrum scrum = new Scrum();
+                    scrum.CreatedOn = DateTime.UtcNow.Date;
+                    scrum.SlackChannelId = slackChannelId;
+                    scrum.ScrumDate = DateTime.UtcNow.Date;
+                    scrum.ProjectId = project.Id;
+                    scrum.TeamLeaderId = project.TeamLeaderId;
+                    scrum.IsHalted = false;
+                    scrum.IsOngoing = true;
+                    _scrumRepository.Insert(scrum);
+                    await _scrumRepository.SaveChangesAsync();
+
+                    //add the scrum details to the temporary table
+                    await AddTemporaryScrumDetailsAsync(scrum.Id, firstUser.SlackUserId, 0, question.Id);
+                    //first user is asked questions along with the previous day status (if any)
+                    replyMessage = string.Format(_stringConstant.GoodDay, slackUserDetailAc.Name) + FetchPreviousDayStatus(firstUser.Id, project.Id, questionList) + question.QuestionStatement + Environment.NewLine;
```

Comment:
slackUserDetailAc could potentially throw a NullReferenceException since you are only checking for it's null condition